scripts/factory-common-install/libexec/app_current_device_vnc.sh

```diff
@@ -0,0 +1,168 @@
+        echo "x11vnc systemd process is already closed"
+    else
+        echo "Closing x11vnc systemd process"
+        close_x11vnc_systemd_process "${vnc_user}" "${vnc_hostname}" "${device_ssh_port}" "${systemd_service_name}"
+    fi
+
+    if is_x11vnc_pid_stopped "${vnc_user}" "${vnc_hostname}" "${device_ssh_port}" "${vnc_remote_port}"; then
+        echo "x11vnc pid is already stopped"
+    else
+        echo "Closing x11vnc pid"
+        exit_x11vnc "${vnc_user}" "${vnc_hostname}" "${device_ssh_port}"
+    fi
+
+    return 0
+  }
+  trap "cleanup '${vnc_user}' '${vnc_hostname}' '${device_ssh_port}' '${vnc_remote_port}' '${systemd_service_name}'" EXIT SIGINT SIGQUIT
+
+  # Validation that the service that we are about to launch does not exist yet.
+  # if the service is alive we ask the user if it wants to kill the existing session.
+  echo " -> Making sure x11vnc is not aready started"
+  if ! is_x11vnc_systemd_process_stopped "${vnc_user}" "${vnc_hostname}" "${device_ssh_port}" "${systemd_service_name}"; then
+    local active_session_info=$(ssh "${vnc_user}@${vnc_hostname}" -p "${device_ssh_port}" "systemctl -q --no-pager --user status '${systemd_service_name}'") && true
+    echo "An active x11vnc session has been detected"
+    echo "Here is some information about the active session"
+    echo "${active_session_info}"
+    echo ""
+    local user_input
+    read -n 1 -p 'Do you want to close the session? (Y)es or (N)o' user_input
+    echo ""
+    case $user_input in
+      [yY] )
+        echo "Okay, we will kill the existing session"
+        cleanup "${vnc_user}" "${vnc_hostname}" "${device_ssh_port}" "${vnc_remote_port}" "${systemd_service_name}"
+        # Sleep 5 seconds to avoid getting kicked out by the other script cleanup.
+        # Todo: Find a proper way to fix the race condition.
+        #       - Perhaps we could stop x11vnc from the new session and wait for the old session to stop the systemd service.
+        sleep 5
+        ;;
+      [nN] )
+        echo "Okay, leaving the existing session as is."
+        trap - EXIT SIGINT SIGQUIT
+        return 0
+        ;;
+      * )
+        echo "Erroneous answer. Doing nothing."
+        trap - EXIT SIGINT SIGQUIT
+        return 0
+        ;;
+    esac
+  fi
+
+  # x11vnc is launched as a connect once server, as soon as vncviewer closes x11vnc also closes.
+  echo " -> Starting x11vnc"
+  local user_info="$(get_factory_info__user_full_name)<$(get_factory_info__user_email)>"
+  local vnc_passwd="$(pwgen -cn 32 | head -n1)"
+  local cmd="$(cat <<EOF
+systemd-run \
+--remain-after-exit \
+--property=Type=forking \
+--unit "${systemd_service_name}" \
+--user bash\
+ -c 'echo "${user_info}"'; x11vnc -bg -once -localhost -passwd "${vnc_passwd}" -display :0
+EOF
+)"
+  if ! ssh "${vnc_user}@${vnc_hostname}" -p "${device_ssh_port}" "${cmd}"; then
+    local err="$?"
```

Comment:
See here having local and right hand side on the same line is perfect become there is no subprocess running on the right hand side that can fail (e.g.: `$(my_sub_process)`).